### PR TITLE
Create wpilib-sys

### PR DESCRIPTION
**Pull request by jmeggitt**
A fix for #9. Might expose a bit too much and I may have incorrectly handled clippy, but it builds.

### CODE REVIEW

File: wpilib/src/lib.rs [hunk 1]
```diff
@@ -4,11 +4,26 @@
...
+pub use self::robot_base::*;
+pub use self::pdp::*;
 
-mod hal;
-pub use hal::{report_usage, report_usage_extras, HalError, HalMaybe, HalResult};
-mod wpilib;
-pub use wpilib::*;
+pub use wpilib_sys::bindings::*;
+pub use wpilib_sys::hal_call::{HalMaybe, HalError, HalResult};
+pub use wpilib_sys::usage::{report_usage, report_usage_extras};
```
Comment (auscompgeek) [inline wpilib/src/lib.rs]: I'm not sure you actually want to reexport these here. You _definitely_ don't want to reexport the FFI bindings.

File: wpilib-sys/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+categories = ["embedded", "hardware-support", "external-ffi-bindings"]
+license = "GPL-3.0"
+
+include = [
+    "src/**/*",
+    "Cargo.toml",
+    "build.rs",
+    "HAL/WPILIB_LICENSE",
+    "HAL/include/**/*",
+    "HAL/lib/**/*",]
```
Comment (auscompgeek) [inline wpilib-sys/Cargo.toml]: You'll want to double-check these includes actually match what you're expecting.

File: wpilib-sys/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+exclude = [".cargo", "./**/makefile"]
+
+[features]
+dev = []
+
+[dependencies]
+lazy_static = "1.0.1"
```
Comment (auscompgeek) [inline wpilib-sys/Cargo.toml]: Where does this depend on `lazy_static`?

File: wpilib-sys/Cargo.toml (hunk 2 above)
Comment (jmeggitt) [reply wpilib-sys/Cargo.toml]: Yea, I never checked these.

File: wpilib/src/lib.rs (hunk 1 above)
Comment (jmeggitt) [reply wpilib/src/lib.rs]: I'm not sure about the usage. It seems like something someone might actually use on their robot, but they have more use in wpilib-sys.

**Comment by jmeggitt** [comment]
This pull request also includes #12.

File: wpilib/src/lib.rs (hunk 1 above)
Comment (auscompgeek) [reply wpilib/src/lib.rs]: If they really want to use it, having them usable only from the `wpilib_sys` crate would make it clear that it's not the intended API. Also we probably don't want to clutter the docs with all the HAL functions.

File: wpilib/src/lib.rs [hunk 4]
```diff
@@ -4,11 +4,24 @@
 // <https://www.gnu.org/licenses/> for a copy.
 
 #![feature(concat_idents)]
-
-#[macro_use]
+#![macro_use]
```
Comment (auscompgeek) [inline wpilib/src/lib.rs]: Hang on. Why was this line changed? You want to use the `lazy_static!` macro here, right?

File: wpilib/src/lib.rs [hunk 5]
```diff
@@ -4,11 +4,24 @@
 // <https://www.gnu.org/licenses/> for a copy.
 
 #![feature(concat_idents)]
```
Comment (auscompgeek) [inline wpilib/src/lib.rs]: Hmm. Is this feature still used in this crate directly? I think this only applies to `wpilib-sys`, unless there's a macro invocation somewhere I haven't noticed.

File: wpilib/src/lib.rs (hunk 4 above)
Comment (jmeggitt) [reply wpilib/src/lib.rs]: It did not compile without the exclamation mark.

File: wpilib/src/lib.rs (hunk 5 above)
Comment (jmeggitt) [reply wpilib/src/lib.rs]: The use of resource_type!() from wpilib-sys in wpilib means it also requires the feature.

File: .dockerignore [hunk 6]
```diff
@@ -20,8 +20,8 @@ cargo-frc/target
 
 !wpilib-examples
 
-!wpilib
```
Comment (auscompgeek) [inline .dockerignore]: I think you still want to not ignore the wpilib directory here.
